code/game/objects/items/weapons/implants/implant_adrenalin.dm

```diff
@@ -0,0 +1,39 @@
+/obj/item/implant/adrenalin
+	name = "adrenal implant"
+	desc = "Removes all stuns and knockdowns."
+	icon_state = "adrenal"
+	origin_tech = "materials=2;biotech=4;combat=3;syndicate=4"
+	uses = 3
+	implant_data = /datum/implant_fluff/adrenaline
+	implant_state = "implant-syndicate"
+
+/obj/item/implant/adrenalin/activate()
+	uses--
+	to_chat(imp_in, "<span class='notice'>You feel a sudden surge of energy!</span>")
+	imp_in.SetStunned(0)
+	imp_in.SetWeakened(0)
+	imp_in.SetKnockDown(0)
+	imp_in.SetParalysis(0)
+	imp_in.adjustStaminaLoss(-75)
+	imp_in.stand_up(TRUE)
+
+	imp_in.reagents.add_reagent("synaptizine", 10)
+	imp_in.reagents.add_reagent("omnizine", 10)
+	imp_in.reagents.add_reagent("stimulative_agent", 10)
+	if(!uses)
+		qdel(src)
```

Comment:
Does this remove it from the user or just leave them with a null item inside? I'm not entirely unconvinced that this might be causing some issues with implant checks.

Reply:
I have no idea, I just moved this to a completely new file, all of this is OG code from before this PR. It's entirely possible that there is a ref list that includes this implant somewhere and there isn't a signal registered on the implant to listen to see if it's QDEL'd

Reply:
Gotcha. I've been suspecting recently that the reason implant checks sometimes seem to need to be done multiple times may have a few causes: the fact that it has no fail step, meaning no message is displayed if it fails (which I'm trying to address in my PR), and the possibility that it is finding a null/expired implant inside the body.